### PR TITLE
Add MaterialData for Dirt

### PR DESCRIPTION
**Pull request by jimmikaelkael**
This adds the possibility to identify the three types of dirt: normal, coarse and podzol.

---

_Edited by turt2live_

**Related links:**
- Glowstone: None needed

### CODE REVIEW

File: src/main/java/org/bukkit/DirtType.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+import java.util.Map;
+
+import com.google.common.collect.Maps;
+
+/**
+ * Represents the three different types of dirt.
+ */
+public enum DirtType {
+
```
Comment (turt2live) [inline src/main/java/org/bukkit/DirtType.java]: Extra line

File: src/main/java/org/bukkit/DirtType.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    /**
+     * Represents the normal dirt.
+     */
+    NORMAL(0x0),
+    /**
+     * Represents the coarse dirt.
+     */
+    COARSE(0x1),
+    /**
+     * Represents the podzol.
```
Comment (turt2live) [inline src/main/java/org/bukkit/DirtType.java]: Please adjust the sentence structure to make more sense

File: src/main/java/org/bukkit/DirtType.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+ * Represents the three different types of dirt.
+ */
+public enum DirtType {
+
+    /**
+     * Represents the normal dirt.
+     */
+    NORMAL(0x0),
+    /**
+     * Represents the coarse dirt.
```
Comment (turt2live) [inline src/main/java/org/bukkit/DirtType.java]: Please adjust the sentence structure to make more sense

File: src/main/java/org/bukkit/DirtType.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+import com.google.common.collect.Maps;
+
+/**
+ * Represents the three different types of dirt.
+ */
+public enum DirtType {
+
+    /**
+     * Represents the normal dirt.
```
Comment (turt2live) [inline src/main/java/org/bukkit/DirtType.java]: Please adjust the sentence structure to make more sense

File: src/main/java/org/bukkit/DirtType.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+    private final byte data;
+    private final static Map<Byte, DirtType> BY_DATA = Maps.newHashMap();
+
+    private DirtType(final int data) {
+        this.data = (byte) data;
+    }
+
+    /**
+     * Gets the associated data value representing this type
```
Comment (turt2live) [inline src/main/java/org/bukkit/DirtType.java]: Missing period

File: src/main/java/org/bukkit/DirtType.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+     *
+     * @return A byte containing the data value of this dirt type
```
Comment (turt2live) [inline src/main/java/org/bukkit/DirtType.java:34]: ~~`@return a byte...` (case)~~

I'm dumb

File: src/main/java/org/bukkit/DirtType.java [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+     * @deprecated Magic value
+     */
+    @Deprecated
+    public byte getData() {
+        return data;
+    }
+
+    /**
+     * Gets the DirtType with the given data value
```
Comment (turt2live) [inline src/main/java/org/bukkit/DirtType.java]: Missing period

File: src/main/java/org/bukkit/DirtType.java [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+     *
+     * @param data Data value to fetch
```
Comment (turt2live) [inline src/main/java/org/bukkit/DirtType.java]: `@param data The data value to fetch`

File: src/main/java/org/bukkit/DirtType.java [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+     * @return The {@link DirtType} representing the given value, or null
```
Comment (turt2live) [inline src/main/java/org/bukkit/DirtType.java:46]: ~~Case (`The` -> `the`)~~

I'm dumb

File: src/main/java/org/bukkit/material/Dirt.java [hunk 10]
```diff
@@ -0,0 +1,77 @@
+package org.bukkit.material;
+
+import org.bukkit.DirtType;
+import org.bukkit.Material;
+
+/**
+ * Represents the different types of dirt.
+ */
+public class Dirt extends MaterialData {
```
Comment (turt2live) [inline src/main/java/org/bukkit/material/Dirt.java:9]: Missing blank line after declaration

File: src/main/java/org/bukkit/material/Dirt.java [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+        super(Material.DIRT);
+    }
+
+    public Dirt(DirtType dirt) {
+        this();
+        setType(dirt);
+    }
+
+    /**
+     *
```
Comment (turt2live) [inline src/main/java/org/bukkit/material/Dirt.java]: Remove excess line

File: src/main/java/org/bukkit/material/Dirt.java [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+    public Dirt(final int type) {
+        super(type);
+    }
+
+    public Dirt(final Material type) {
+        super(type);
+    }
+
+    /**
+     *
```
Comment (turt2live) [inline src/main/java/org/bukkit/material/Dirt.java]: Remove excess line

File: src/main/java/org/bukkit/material/Dirt.java [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+     * @deprecated Magic value
+     */
+    @Deprecated
+    public Dirt(final int type, final byte data) {
+        super(type, data);
+    }
+
+    /**
+     *
```
Comment (turt2live) [inline src/main/java/org/bukkit/material/Dirt.java]: Remove excess line

File: src/main/java/org/bukkit/material/Dirt.java [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+     * @deprecated Magic value
+     */
+    @Deprecated
+    public Dirt(final Material type, final byte data) {
+        super(type, data);
+    }
+
+    /**
+     * Gets the current type of this dirt
```
Comment (turt2live) [inline src/main/java/org/bukkit/material/Dirt.java]: Missing period

File: src/main/java/org/bukkit/material/Dirt.java [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+     *
+     * @return DirtType of this dirt
```
Comment (turt2live) [inline src/main/java/org/bukkit/material/Dirt.java]: `@return The type of dirt`

File: src/main/java/org/bukkit/material/Dirt.java [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+     */
+    public DirtType getType() {
+        return DirtType.getByData(getData());
+    }
+
+    /**
+     * Sets the type of this dirt
```
Comment (turt2live) [inline src/main/java/org/bukkit/material/Dirt.java]: Missing period

File: src/main/java/org/bukkit/material/Dirt.java [hunk 17]
```diff
@@ -0,0 +1,77 @@
...
+     *
+     * @param dirt New type of this dirt
```
Comment (turt2live) [inline src/main/java/org/bukkit/material/Dirt.java]: `@param dirt The new type of dirt` 

Also consider a parameter name change to something like `type`

File: src/main/java/org/bukkit/DirtType.java (hunk 2 above)
Comment (jimmikaelkael) [inline src/main/java/org/bukkit/DirtType.java]: @turt2live I've tried to do my best, but not being a native english speaker this maybe still not good. Don't hesitate to suggest me a better sentence.

**Comment by jimmikaelkael** [comment]
@turt2live rebased on top of master and made the requested format fixes.

File: src/main/java/org/bukkit/DirtType.java (hunk 2 above)
Comment (turt2live) [inline src/main/java/org/bukkit/DirtType.java]: Your changes are much better. Thank you! 

And I didn't realize English wasn't your first language, I just thought that these were just mishaps in typing. Great job on the English skills I suppose ;)

**Comment by SpaceManiac** [comment]
Merged in 267a795149a14250209b822ec99804c335150a56.

**Pull request closed without merging**